### PR TITLE
fix(status): detect running turns across all omp composer shapes

### PR DESCRIPTION
**Pull request by larkinwc**
## Description
Agent of Empires status detection for OMP (`omp.toml`) fails to detect active turns when OMP is configured with non-box composer shapes (e.g. `composer.shape = claude`, `rule`, `pi`, `borderless`, `field`, `rail`). Sessions stay reported as `Idle` even while actively executing tools or compaction.

In non-box composer shapes, OMP pushes the activity band above the input prompt (`position = 4`) and the interrupt hint (`⎋ <verb>`) above that (`position = 5+`), while parking a separate status line at the very bottom (`position = 1`):
```
  ⎋ Capturing live sessions
───── ⚙ 1 · Review PR · ⏱ 28.6s ─
❯
───────────────────────────────────
 ⠏ 28s · 🖥 host · 🏃 Prewalk
```

The existing `active_brand` rule required the activity band at `max_position = 2` and its interrupt row within `bottom_non_empty_lines(4)`. Furthermore, PR #3788's proposed `active_composer` rule only targeted Unicode box border delimiters (`╭─...╮`), which non-box shapes do not render.

### Solution
Add an `active_statusline` rule in `src/tmux/detect/manifests/omp.toml`:
1. Matches the leading live braille spinner and turn timer (`[\x{2800}-\x{28FF}] <timer>`) at the bottom of the pane (`region = "bottom_non_empty_lines(3)"`).
2. Gated with an `all` requirement on an interrupt row (`⎋`, `esc`, `working…`) or compaction hint (`(esc to cancel)`) within `bottom_non_empty_lines(8)` to ensure stale transcript timers do not trigger false-positive `Running` states when a session is idle.
3. Protected with `not` guards against clock glyphs and nearby bordered waiting/approval panels (`│ up/down navigate enter select esc cancel │`), ensuring `Waiting` states are never masked.

## PR Type

- [ ] New Feature
- [x] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Infrastructure / CI

## Checklist

- [x] New and existing tests pass
- [ ] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording

## Test Coverage Analysis

**Web dashboard / structured view (Playwright user story)**
- [x] N/A: this PR doesn't change a user-facing dashboard flow
- [ ] Added or updated a Playwright spec under `web/tests/` and updated `web/tests/coverage-matrix.json`
- [ ] Skipped a test, because: 

**TUI / CLI (e2e test)**
- [x] N/A: this PR doesn't change TUI rendering, a CLI subcommand, or session lifecycle
- [ ] Added or updated an e2e test under `tests/e2e/`
- [ ] Skipped a test, because: 

Added unit regression tests in `src/tmux/status_detection.rs`:
- `test_detect_omp_status_running_on_active_statusline` covering all composer shapes (`claude`, `rule`, `pi`, `borderless`, `field`, `rail`, `band`) and auto server compaction.
- Near-miss idle tests (stale band with parked `π` footer, parked claude shape at prompt) and lowest-marker tests ensuring waiting dialogs win over spinning statuslines.

## AI Usage

- [ ] No AI was used
- [x] AI was used

**AI Model/Tool used:**
Oh My Pi coding harness / gemini-3.8-flash

**Any Additional AI Details you'd like to share:**
AI was used to investigate the root cause across OMP composer layouts, design the manifest rule with positional arbitration and waiting guards, and add regression tests covering all composer shapes.

**NOTE:**
When responding to reviewer questions, please respond yourself rather than copy/pasting reviewer comments into an AI and pasting back its answer. We want to discuss with you, not your AI :) 

- [x] I am an AI Agent filling out this form (check box if true)

## Summary

- Improved OMP running-status detection across all composer shapes.
- Added active statusline detection for live spinners and turn timers.
- Reduced false positives from stale timers, waiting dialogs, approval panels, and quoted text.
- Added regression tests for composer shapes, auto-compaction, idle states, dialogs, and statusline edge cases.

## Benefits

OMP status detection now reports `Running` more reliably and avoids inactive or misleading UI indicators.

## Technical Notes

- Added the `active_statusline` rule.
- Restricted cancellation guards to rows with active spinner glyphs.
- Required panel-specific footer structures before suppressing active status.
- Updated test fixtures with neutral hostnames.

## Potential Enhancement

Add coverage for additional statusline layouts as the OMP interface evolves.

### CODE REVIEW

File: src/tmux/status_detection.rs [hunk 1]
```diff
@@ -4416,6 +4506,11 @@ Final prose line.\n";
                 format!("{approval}\n{band}\n╰─"),
                 Status::Running,
             ),
+            (
+                "lower approval wins over active statusline",
+                format!("{approval}\n❯\n───────────────────────────────────\n ⠏ 28s · 🖥 host"),
```
Comment (jerome-benoit) [inline src/tmux/status_detection.rs]: [P3, nonblocking] Exercise the competing active rule in this fixture

This pane has no interrupt or compaction row, so `active_statusline` already fails its mandatory activity guard (`omp.toml:78-84`), even without the dialog exclusion. The assertion therefore cannot catch a regression in that exclusion. Please prepend a valid activity row, e.g. `⎋ Running tests`, within the eight-line window so the bottom spinner would otherwise report Running. This is a coverage gap, not a demonstrated production failure. Evidence: inspected the fixture and manifest at 676f2005 with `git show`, including the matcher's mandatory `all` evaluation in `manifest.rs:311`.

**Review by jerome-benoit — APPROVED** [review]
Reviewed 676f2005 with two independent sequential analyses and cross-validation. No blocking defect established in this two-file change. The resolved CodeRabbit findings were considered; the separate statusline detection complements #3788's boxed multiline-composer fix.

One nonblocking P3 test-coverage note below. Validation: static inspection and `git diff --check d9e5060e 676f2005` passed; no local tests were run. CI was authorized and is still in progress, so this approval does not attest to a completed green test suite.

File: src/tmux/status_detection.rs (hunk 1 above)
Comment (larkinwc) [reply src/tmux/status_detection.rs]: @jerome-benoit went ahead and addressed that

**Review by jerome-benoit — APPROVED** [review]
Re-reviewed all changes since my previous approval at `676f2005`, including `d142dd82`, the imported `89f931d0` (#3760), and merge `7eaeb14c`, with two independent sequential analyses and cross-validation. No new actionable findings.

The P3 is addressed: `status_detection.rs:4511` now supplies the activity signal needed to exercise the dialog exclusion. The imported HOME tests (`server/api/git.rs:455`, `tui/components/preview.rs:617`) retain their assertions, use the shared environment lock, and restore HOME before deleting the temporary directory. Parent-tree comparisons confirm the merge preserves both sides' changes; the OMP manifest is unchanged.

Evidence: exact-head `git show`, `git log 676f2005..7eaeb14c`, full delta inspection and `git diff --check` (exit 0). Current CI, Tests and CodeQL are green at `7eaeb14c`, verified with `gh pr checks 3808`. No local tests were run.
